src/py42/_internal/clients/alertrules.py

```diff
@@ -0,0 +1,156 @@
+              'cloudshare' implies rules related to sharing/providing public access to files
+              'typemismatch' implies rules related to content and file extension mismatch.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {"tenantId": tenant_id, "ruleIds": [rule_id]}
+        uri = u"{0}{1}".format(self._api_prefix, _ALERT_RULE_ENDPOINTS[rule_type])
+        return self._session.post(uri, data=json.dumps(data))
+
+
+class AlertRulesManagerClient(BaseClient):
+    """A client to manage Alert Rules."""
+
+    _version = u"v1"
+    _resource = u"rules/"
+    _api_prefix = u"/svc/api/{0}/{1}".format(_version, _resource)
+
+    def __init__(self, session, user_context):
+        super(AlertRulesManagerClient, self).__init__(session)
+        self._user_context = user_context
+
+    def _get_alert_rules(
+        self, tenant_id, sort_key=None, sort_direction=None, page_num=None, page_size=None
+    ):
+        data = {
+            "tenantId": tenant_id,
+            "groups": [],
+            "groupClause": "AND",
+            "pgNum": 0,  # This API expects first page to start with zero, hence hard-coded.
+            "pgSize": page_size,
+            "srtKey": sort_key,
+            "srtDirection": sort_direction,
+        }
+        uri = u"{0}{1}".format(self._api_prefix, u"query-rule-metadata")
+        return self._session.post(uri, data=json.dumps(data))
+
+    def get_all(self, sort_key="CreatedAt", sort_direction="DESC"):
+        """Fetch all available rules.
+
+        Args:
+            sort_key (str): Sort results based by field, default 'CreatedAt'.
+            sort_direction (str): ``ASC`` or ``DESC``, default "DESC"
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response` objects
+            that each contain a page of events.
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        return get_all_pages(
+            self._get_alert_rules,
+            u"ruleMetadata",
+            tenant_id=tenant_id,
+            sort_key=sort_key,
+            sort_direction=sort_direction,
+        )
+
+    def get_by_name(self, rule_name):
+        """Fetch a rule by its name.
+
+            Raises :class:`py42.exceptions.Py42NotFoundError` when no match is found.
+        Args:
+            rule_name (str): Rule name to search for, case insensitive search.
+
+        Returns
+            :dict: Dictionary containing rule-details.
+        """
+        rule_pages = self.get_all()
+        for rule_page in rule_pages:
+            rules = rule_page["ruleMetadata"]
+            for rule in rules:
+                if rule_name.lower() in rule["name"].lower():
+                    return rule
+        raise Py42NotFoundError("No Alert Rules found with name: {0}".format(rule_name))
```

Comment:
I think creating this Exception will fail on the latest version of py42 -- we updated all of the HTTP-based error to take the root exception in the constructor instead of a message. We can raise a generic `Py42Error` here.

Reply:
Changed to Py42Error with the message.